sormas-ui/src/main/java/de/symeda/sormas/ui/utils/CheckBoxTreeUpdated.java

```diff
@@ -0,0 +1,249 @@
+package de.symeda.sormas.ui.utils;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+import java.util.function.UnaryOperator;
+
+import org.jetbrains.annotations.NotNull;
+
+import com.vaadin.server.Sizeable;
+import com.vaadin.shared.ui.MarginInfo;
+import com.vaadin.ui.Component;
+import com.vaadin.ui.HorizontalLayout;
+import com.vaadin.ui.Label;
+import com.vaadin.ui.VerticalLayout;
+import com.vaadin.v7.data.util.converter.Converter;
+import com.vaadin.v7.ui.CheckBox;
+import com.vaadin.v7.ui.CustomField;
+
+public class CheckBoxTreeUpdated<ENUM extends Enum<?>> extends CustomField<Map<ENUM, Boolean>> {
+
+	private static final String[] INDENTATION_STYLES = new String[] {
+		CssStyles.INDENT_LEFT_1,
+		CssStyles.INDENT_LEFT_2,
+		CssStyles.INDENT_LEFT_3 };
+
+	private Map<ENUM, CheckBox> enumToggles = new HashMap<>();
+	private VerticalLayout checkBoxTreeLayout;
+	private HorizontalLayout checkBoxTreeLayoutForGroups;
+	private Class<ENUM> enumType;
+	private boolean addVerticalSpaces;
+	private UnaryOperator<ENUM> parentProvider;
+	private Function<ENUM, ? extends Enum> groupProvider;
+	private Class<? extends Enum> parentGroup;
+	private boolean settingToggles = false;
+	private int columns;
+
+	public CheckBoxTreeUpdated() {
+
+	}
+
+	public Map<ENUM, CheckBox> getEnumToggles() {
+		return enumToggles;
+	}
+
+	@Override
+	protected Component initContent() {
+
+		if (enumType != null) {
+			if (groupProvider == null) {
+				checkBoxTreeLayout = new VerticalLayout();
+				checkBoxTreeLayout.setMargin(false);
+				checkBoxTreeLayout.setSpacing(false);
+				checkBoxTreeLayout.setWidth(100, Sizeable.Unit.PERCENTAGE);
+				addCheckBoxes();
+				final MarginInfo marginInfo = new MarginInfo(false, false, true, false);
+				checkBoxTreeLayout.setMargin(marginInfo);
+			} else {
+				checkBoxTreeLayoutForGroups = new HorizontalLayout();
```

Comment:
rename to `groupLayout` and the othe one to `checkboxLayout` to be shorter

Reply:
renamed